root/components/list/RecordingList.js

```diff
@@ -143,11 +145,22 @@ const RecordingList = ({
             {/* Show nothing rather than ?:?? for recordings merged away */}
             {recording.gid ? formatTrackLength(recording.length) : null}
           </td>
-          {showInstrumentCredits ? (
+          {showInstrumentCreditsAndRelTypes ? (
             <td>
-              {instrumentCredits && instrumentCredits[recording.gid]
-                ? commaListText(instrumentCredits[recording.gid])
-                : null}
+              {instrumentCreditsAndRelTypes &&
+                instrumentCreditsAndRelTypes[recording.gid] ? (
+                  commaOnlyListText(
+                    instrumentCreditsAndRelTypes[recording.gid].map(json => {
+                      const relType = JSON.parse(json);
+                      let listElement = l_relationships(relType.name);
+                      if (relType.credit) {
+                        listElement = listElement + ' ' +
+                          bracketedText(texp.l('as “{credit}”', {credit: relType.credit}));
+                      }
+                      return listElement;
+                    }),
+                  )
+                ) : null}
```

Comment:
It'd be nice to factor this out into a function that can be shared with root/components/list/ReleaseList.js, since it is kinda complex enough that we might want to make changes/fixes in both places at some point.